rules/terraformrules/terraform_empty_list_equality.go

```diff
@@ -55,27 +56,37 @@ func (r *TerraformEmptyListEqualityRule) Check(runner *tflint.Runner) error {
 func (r *TerraformEmptyListEqualityRule) checkEmptyList(runner *tflint.Runner) error {
 	return runner.WalkExpressions(func(expr hcl.Expression) error {
 		if conditionalExpr, ok := expr.(*hclsyntax.ConditionalExpr); ok {
-			if binaryOpExpr, ok := conditionalExpr.Condition.(*hclsyntax.BinaryOpExpr); ok {
-				if binaryOpExpr.Op.Type.FriendlyName() == "bool" {
-					if right, ok := binaryOpExpr.RHS.(*hclsyntax.TupleConsExpr); ok {
-						checkEmptyList(right, runner, r, binaryOpExpr)
-					}
-					if left, ok := binaryOpExpr.LHS.(*hclsyntax.TupleConsExpr); ok {
-						checkEmptyList(left, runner, r, binaryOpExpr)
-					}
-				}
-			}
+			var issuesRangeSet = make(map[hcl.Range]void)
+			searchEmptyList(conditionalExpr.Condition, runner, r, conditionalExpr.Range(), issuesRangeSet)
```

Comment:
Making your helper functions into methods on `*TerraformEmptyListEqualityRule` would eliminate the naming issues

Reply:
made it a method but kept the name. Also added comments to all the `*TerraformEmptyListEqualityRule` methods to be more clear on what they're doing